Framework/WorkflowAlgorithms/src/AlignAndFocusPowder.cpp

```diff
@@ -239,6 +257,31 @@ std::map<std::string, std::string> AlignAndFocusPowder::validateInputs() {
                                         "Please plot the pCharge-time to identify suitable range for "
                                         "re-time-slicing";
 
+  m_resonanceLower = getProperty(PropertyNames::RESONANCE_LOWER_LIMITS);
+  m_resonanceUpper = getProperty(PropertyNames::RESONANCE_UPPER_LIMITS);
+  if (m_resonanceLower.size() > 0 || m_resonanceUpper.size() > 0) {
```

Comment:
I don't think you need this if-statement.  If either of them is zero or both of them are zero, the codes in the condition block shall handle it well.